NineChronicles.Headless/GraphTypes/StateQuery.cs

```diff
@@ -18,42 +19,106 @@ public StateQuery()
             Name = "StateQuery";
             Field<AvatarStateType>(
                 name: "avatar",
-                arguments: new QueryArguments(new QueryArgument<AddressType>
-                {
-                    Name = "address",
-                }),
+                arguments: new QueryArguments(
+                    new QueryArgument<AddressType>
+                    {
+                        Name = "address",
+                    },
+                    new QueryArgument<ByteStringType>
+                    {
+                        Name = "hash",
+                        Description = "Offset block hash for query."
+                    }),
                 resolve: context =>
                 {
                     var address = context.GetArgument<Address>("address");
-                    return new AvatarState((Dictionary)context.Source.GetState(address));
+                    byte[] blockHashByteArray = context.GetArgument<byte[]>("hash");
```

Comment:
Perhaps, we may need to add a primitive type for `HashDigest<SHA256>` like `Address` to avoid duplications 🤔